### DISCUSSION TITLE
chainlink mix for vrf deploy error: Unable to expand environment variable in host setting: 'https://kovan.infura.io/v3/$WEB3_INFURA_PROJECT_ID'

### DISCUSSION THREAD

**Discussion by ptn77** [smartcontractkit/full-blockchain-solidity-course-py#1383] [Q&A] [answered]
Hi,
I am trying to deploy the 01_deploy_vrf.py from the vrf chainlink mix.
I updated the .env.examples file with the valid values and changed the file name to .env
I tried adding environment values with and without the quotes, but both gave the same error. 
Does anyone know what I'm doing wrong? Should the .env file exist in another directory or folder within the MIXES folder?

Then I ran brownie run scripts/vrf_scripts/01_deploy_vrf.py --network kovan

I'm getting the following error: 
```
ChainlinkProject is the active project.
  File "brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "brownie/_cli/run.py", line 43, in main
    network.connect(CONFIG.argv["network"])
  File "brownie/network/main.py", line 40, in connect
    web3.connect(host, active.get("timeout", 30))
  File "brownie/network/web3.py", line 52, in connect
    uri = _expand_environment_vars(uri)
  File "brownie/network/web3.py", line 183, in _expand_environment_vars
    raise ValueError(f"Unable to expand environment variable in host setting: '{uri}'")
ValueError: Unable to expand environment variable in host setting: 'https://kovan.infura.io/v3/$WEB3_INFURA_PROJECT_ID'

```

thanks

**Comment by ptn77** [comment, marked as answer]
figured it out.
I needed to put dotenv : .env in the brownie-config.yaml file

**Reply by cromewar** [reply]
Hello @ptn77 sorry for the delay, I'm glad you solved this by yourself.
Feel free to make any other question on the repo, cheers.
